### PR TITLE
Modify --use-param-defaults flag for `pipeline start` command and add e2e test

### PR DESCRIPTION
**Pull request by Divyansh42**
# Changes

Modified --use-param-defaults flag for `pipeline start` command
to not add `default params` to pipelinerun which should be automatically
added by the controller itself and prompt interactive mode if default is not
provided for some params in the pipeline and also added `e2e tests` for 
the `--use-param-defaults` option.

Signed-off-by: Divyansh42 <diagrawa@redhat.com>

Fixes: #1065 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes
```release-note
Modify --use-param-defaults flag for pipeline start command to prompt interactive mode if default is not provided for some params.
```

### CODE REVIEW

File: pkg/cmd/pipeline/start.go [hunk 1]
```diff
@@ -358,38 +358,6 @@ func (opt *startOptions) getInput(pipeline *v1beta1.Pipeline) error {
 		}
 	}
 
-	if opt.UseParamDefaults && !opt.Last && opt.UsePipelineRun == "" {
-		if err = opt.includeDefaultParams(pipeline); err != nil {
-			return err
-		}
-	}
-
-	return nil
```
Comment (piyush-garg) [inline pkg/cmd/pipeline/start.go:367]: I think we need this

File: test/builder/builder.go [hunk 2]
```diff
@@ -641,7 +641,7 @@ const describeTemplate = `{{decorate "bold" "Name"}}:	{{ .PipelineName }}
  NAME	TYPE	DESCRIPTION	DEFAULT VALUE
 {{- range $i, $p := .Pipeline.Spec.Params }}
 {{- if not $p.Default }}
- {{decorate "bullet" $p.Name }}	{{ $p.Type }}	{{ $p.Description }}	{{ "" }}
```
Comment (piyush-garg) [inline test/builder/builder.go:644]: Is not the same change required in cmd also then

File: pkg/cmd/pipeline/start.go (hunk 1 above)
Comment (Divyansh42) [reply pkg/cmd/pipeline/start.go:367]: We have discussed this thing in PR #1036 to proceed with the same way we have done for task start [here](https://github.com/tektoncd/cli/blob/master/pkg/cmd/task/start.go#L530) and correct that in pipeline start command.

File: test/builder/builder.go (hunk 2 above)
Comment (Divyansh42) [reply test/builder/builder.go:644]: If I have understood that right, it's already there. Can you please check once [here](https://github.com/tektoncd/cli/blob/master/pkg/cmd/pipeline/describe.go#L63)

File: pkg/cmd/pipeline/start.go (hunk 1 above)
Comment (piyush-garg) [reply pkg/cmd/pipeline/start.go:367]: I am saying for just one line `return nil` not the whole block, I see it there in the above-mentioned PR also

File: test/builder/builder.go (hunk 2 above)
Comment (piyush-garg) [reply test/builder/builder.go:644]: Ok, so this was missed here. LGTM

File: pkg/cmd/pipeline/start.go (hunk 1 above)
Comment (Divyansh42) [reply pkg/cmd/pipeline/start.go:367]: Okay, will add this. That was by mistake :sweat_smile: , thanks for the review :smile:

**Review by piyush-garg — APPROVED** [review]

**Comment by danielhelfand** [comment]
@Divyansh42 You will need to rebase and update the `Start PipelineRun with --pod-template` to add the `--use-param-defaults` to the command used in the test since this pr updates the `output-pipeline` used in the test.

https://github.com/tektoncd/cli/blob/ca357f8b8ffa297a48f94302f13533a555418578/test/e2e/pipeline/pipeline_test.go#L319

**Comment by danielhelfand** [comment]
These test failures aren't flakes. Rerunning them will not work until the [comment above](https://github.com/tektoncd/cli/pull/1109#issuecomment-680013998) is addressed.

**Comment by Divyansh42** [comment]
> @Divyansh42 You will need to rebase and update the `Start PipelineRun with --pod-template` test to add the `--use-param-defaults` to the command used in the test since this pr updates the `output-pipeline` used in the test.
> 
> https://github.com/tektoncd/cli/blob/ca357f8b8ffa297a48f94302f13533a555418578/test/e2e/pipeline/pipeline_test.go#L319

Will rebase and update the PR.

**Comment by Divyansh42** [comment]
@danielhelfand Updated the e2e tests.

**Comment by danielhelfand** [comment]
@Divyansh42 Please just squash your commits into a single on, and then I think this looks good.

**Pull request merged**